### PR TITLE
Init End To End test using venom

### PR DESCRIPTION
**Pull request by olblak**
Signed-off-by: Olblak <me@olblak.com>

# Add End To End tests

The goal of those tests is to better test different scenarios and to be sure that updatecli pipeline configurations is still working

## Test

To test this pull request, you can run the following commands from a Linux machine:

```shell
# export VENOM_VAR_binpath=<path to you updatecli binary directory>
make build
make test-e2e
```

### Tradeoff

As a first iteration, I am testing that a pipeline configuration doesn't contain ERROR or WARNING.
Considering that I would like to land this pr as soon as possible so we can add more test cases later, I am not updating obsolete examples (yet).

### Potential improvement
Move obsolete examples in the e2e  directory

### CODE REVIEW

**Comment by olblak** [comment]
@dduportal End to end tests are green \o/
So I propose to merge this pull request and then migrate examples from directory "examples" to `e2e` in a second pullrequest.

**Comment by dduportal** [comment]
Haven't tested extensively yet, but the checks are failing because of the venom's manifest for info

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,9 @@ DOCKER_TAG=$(shell git describe --tags)
 DOCKER_BUILDKIT=1
 export DOCKER_BUILDKIT
 
+VENOM_VAR_binpath=$(PWD)/dist/updatecli_linux_amd64
```
Comment (dduportal) [inline Makefile:10]:
```suggestion
VENOM_VAR_binpath=$(CURDIR)/dist/updatecli_$(shell go env GOOS)_$(shell go env GOPATH)
```

**Review by dduportal — CHANGES_REQUESTED** [review]
I'm sorry but I don't understand how to use venom locally.

On my machine, it fails with the message

```
venom run e2e/venom.d/test_version.yaml --output-dir ./e2e --format yaml --verbose
          [trac] writing e2e/venom.log
 • Updatecli Version TestSuite (e2e/venom.d/test_version.yaml)
        • Test-updatecli-version-return-code-is-0-binpath-updatecli FAILURE
Testcase "Test updatecli version return code is 0 {{ .binpath }}/updatecli", step #0: Assertion "result.code ShouldEqual 0" failed. expected: 0  got: 127 (e2e/venom.d/test_version.yaml:13)
Testcase "Test updatecli version return code is 0 {{ .binpath }}/updatecli", step #0: Assertion "result.systemerr ShouldContainSubstring \"VERSION\"" failed. expected '/var/folders/2s/09szbrgn22l2tcvxz1_k34g40000gn/T/venom-3683137089: line 1: {{: command not found' to contain 'VERSION' but it wasn't (e2e/venom.d/test_version.yaml:16)
Testcase "Test updatecli version return code is 0 {{ .binpath }}/updatecli", step #0: Assertion "result.systemerr ShouldContainSubstring \"Application:\"" failed. expected '/var/folders/2s/09szbrgn22l2tcvxz1_k34g40000gn/T/venom-3683137089: line 1: {{: command not found' to contain 'Application:' but it wasn't (e2e/venom.d/test_version.yaml:17)
Testcase "Test updatecli version return code is 0 {{ .binpath }}/updatecli", step #0: Assertion "result.systemerr ShouldContainSubstring \"Golang     :\"" failed. expected '/var/folders/2s/09szbrgn22l2tcvxz1_k34g40000gn/T/venom-3683137089: line 1: {{: command not found' to contain 'Golang     :' but it wasn't (e2e/venom.d/test_version.yaml:18)
Testcase "Test updatecli version return code is 0 {{ .binpath }}/updatecli", step #0: Assertion "result.systemerr ShouldContainSubstring \"Build Time :\"" failed. expected '/var/folders/2s/09szbrgn22l2tcvxz1_k34g40000gn/T/venom-3683137089: line 1: {{: command not found' to contain 'Build Time :' but it wasn't (e2e/venom.d/test_version.yaml:19)
Writing file e2e/test_results.yaml
```

I don't understand how it can works on the checks: it's a blocker for merging because any error on the CI would block any contributor as we could not test locally.

Do you mind adding a bit of documentation also ?

File: Makefile (hunk 1 above)
Comment (dduportal) [reply Makefile:10]: This was the cause. Might need documentation

Comment (dduportal) [reply Makefile:10]: Also, missing `GITHUB_TOKEN` and `GITHUB_ACTOR` environment variables

**Review by dduportal — APPROVED** [review]
OK, I was able to make it work locally. It needs documentation, but I propose to do it on a subsequent PR.

File: Makefile (hunk 1 above)
Comment (dduportal) [reply Makefile:10]: Makefile fixed in https://github.com/updatecli/updatecli/pull/457

**Pull request merged**